cmd/config/config.go

```diff
@@ -1,82 +1,178 @@
 package main
 
 import (
-	"encoding/json"
+	"errors"
 	"flag"
-	"log"
+	"fmt"
 	"os"
+	"slices"
+	"strings"
 
+	"github.com/buger/jsonparser"
 	"github.com/thrasher-corp/gocryptotrader/common/file"
 	"github.com/thrasher-corp/gocryptotrader/config"
 )
 
-// EncryptOrDecrypt returns a string from a boolean
-func EncryptOrDecrypt(encrypt bool) string {
-	if encrypt {
-		return "encrypted"
-	}
-	return "decrypted"
-}
+var commands = []string{"upgrade", "encrypt", "decrypt"}
 
 func main() {
-	var inFile, outFile, key string
-	var encrypt bool
+	fmt.Println("GoCryptoTrader: config-helper tool")
+
 	defaultCfgFile := config.DefaultFilePath()
-	flag.StringVar(&inFile, "infile", defaultCfgFile, "The config input file to process.")
-	flag.StringVar(&outFile, "outfile", defaultCfgFile+".out", "The config output file.")
-	flag.BoolVar(&encrypt, "encrypt", true, "Whether to encrypt or decrypt.")
-	flag.StringVar(&key, "key", "", "The key to use for AES encryption.")
-	flag.Parse()
-
-	log.Println("GoCryptoTrader: config-helper tool.")
-
-	if key == "" {
-		result, err := config.PromptForConfigKey(false)
-		if err != nil {
-			log.Fatalf("Unable to obtain encryption/decryption key: %s", err)
-		}
-		key = string(result)
+
+	var in, out, keyStr string
+	var inplace bool
+
+	fs := flag.NewFlagSet("config", flag.ExitOnError)
+	fs.Usage = func() { usage(fs) }
+	fs.StringVar(&in, "in", defaultCfgFile, "The config input file to process")
+	fs.StringVar(&out, "out", "[in].out", "The config output file")
+	fs.BoolVar(&inplace, "edit", false, "Edit; Save result to the original file")
+	fs.StringVar(&keyStr, "key", "", "The key to use for AES encryption")
+
+	cmd, args := parseCommand(os.Args[1:])
+	if cmd == "" {
+		usage(fs)
+		os.Exit(2)
+	}
+
+	if err := fs.Parse(args); err != nil {
+		fatal(err.Error())
+	}
+
+	if inplace {
+		out = in
+	} else if out == "[in].out" {
+		out = in + ".out"
+	}
+
+	key := []byte(keyStr)
+	var err error
+	switch cmd {
+	case "upgrade":
```

Comment:
I can't replicate :'(
The first failure was harmless. It errored, didn't rewrite, and everything should be good.
I can run encrypt, upgrade over and over, and it's always encrypted afterwards.

Stupid questions, because I don't know what else to check:
* Is there anything omitted from this shell transcript?
* Anything might have happened in another terminal?
* Can you replicate again?